inference-engine/ie_bridges/java/cpp/tensor_desc.cpp

```diff
@@ -0,0 +1,196 @@
+    }
+    return 0;
+}
+
+JNIEXPORT void JNICALL Java_org_intel_openvino_TensorDesc_setDims(JNIEnv *env, jobject, jlong addr, jintArray dims){
+    static const char method_name[] = "setDims";
+    try
+    {
+        TensorDesc *tDesc = (TensorDesc *)addr;
+
+        const jsize length = env->GetArrayLength(dims);
+        jint *data = env->GetIntArrayElements(dims, 0);
+
+        std::vector<size_t> sVector;     
+        for (int i = 0; i < length; i++)
+            sVector.push_back(static_cast<size_t>(data[i]));
+
+        tDesc->setDims(sVector);
+        
+    } catch (const std::exception &e)
+    {
+        throwJavaException(env, &e, method_name);
+    }
+    catch (...)
+    {
+        throwJavaException(env, 0, method_name);
+    }
+}
+
+JNIEXPORT jintArray JNICALL Java_org_intel_openvino_TensorDesc_GetDims(JNIEnv *env, jobject obj, jlong addr)
+{
+    static const char method_name[] = "GetDims";
+
+    try
+    {
+        TensorDesc *tDesk = (TensorDesc *)addr;
+        std::vector<size_t> size_t_dims = tDesk->getDims();
+
+        jclass intArray = env->FindClass("[I");
+        if (intArray == nullptr)
+            return nullptr;
+
+        jintArray result = env->NewIntArray(size_t_dims.size());
+        jint *arr = env->GetIntArrayElements(result, nullptr);
+
+        for (int i = 0; i < size_t_dims.size(); ++i)
+        {
+            arr[i] = size_t_dims[i];
+        }
+
+        env->ReleaseIntArrayElements(result, arr, 0);
+
+        return result;
+    }
+    catch (const std::exception &e)
+    {
+        throwJavaException(env, &e, method_name);
+    }
+    catch (...)
+    {
+        throwJavaException(env, 0, method_name);
+    }
+
+    return 0;
+}
+
+JNIEXPORT void JNICALL Java_org_intel_openvino_TensorDesc_setLayout(JNIEnv *env, jobject, jlong addr, jint layout){
+    static const char method_name[] = "setLayout";
+    try
+    {
+        TensorDesc *tDesk = (TensorDesc *)addr;
+        auto it = layout_map.find(layout_e(layout));
+
+        if (it == layout_map.end())
+            throw std::runtime_error("No such layout value!");
+
+        tDesk->setLayout(it->second);
+
+    } catch (const std::exception &e){
+        throwJavaException(env, &e, method_name);
+    }
+    catch (...)
+    {
+        throwJavaException(env, 0, method_name);
+    }
+}
+
+JNIEXPORT jobject JNICALL Java_org_intel_openvino_TensorDesc_getLayout(JNIEnv *env, jobject, jlong addr){
+    static const char method_name[] = "getLayout";
+    try
+    {
+        TensorDesc *tDesk = (TensorDesc *)addr;
+        Layout layout = tDesk->getLayout();
+
+        auto layout_res = find_by_value(layout_map, layout);
+
+        jclass layout_class = env->FindClass("org/intel/openvino/Layout");
+        jmethodID valueOf_method_id = env->GetStaticMethodID(layout_class, "valueOf", "(I)Lorg/intel/openvino/Layout;");
+        jobject layout_object = env->CallStaticObjectMethod(layout_class, valueOf_method_id, layout_res);
+
+        return layout_object;
```

Comment:
Can we just return `layout_res`? Just `static_cast<jobject>(layout_res)` or `static_cast<jint>(layout_res)`?